pkg/imgpkg/internal/util/progress_logger.go

```diff
@@ -14,32 +14,40 @@ import (
 	"github.com/mattn/go-isatty"
 )
 
+// ProgressLogger Progress bar
 type ProgressLogger interface {
-	Start(progress <-chan regv1.Update)
+	Start(ctx context.Context, progress <-chan regv1.Update)
 	End()
 }
 
 // NewProgressBar constructor to build a ProgressLogger responsible for printing out a progress bar using updates when
 // writing to a registry via ggcr
 func NewProgressBar(ui goui.UI, finalMessage, errorMessagePrefix string) ProgressLogger {
-	ctx, cancel := context.WithCancel(context.Background())
 	if isatty.IsTerminal(os.Stdout.Fd()) {
-		return &ProgressBarLogger{ctx: ctx, cancelFunc: cancel, ui: ui, finalMessage: finalMessage, errorMessagePrefix: errorMessagePrefix}
+		return &ProgressBarLogger{ui: ui, finalMessage: finalMessage, errorMessagePrefix: errorMessagePrefix}
 	}
 
-	return &ProgressBarNoTTYLogger{ui: ui, ctx: ctx, cancelFunc: cancel, finalMessage: finalMessage}
+	return &ProgressBarNoTTYLogger{ui: ui, finalMessage: finalMessage}
 }
 
+// NewNoopProgressBar constructs a Noop Progress bar that will not display anything
+func NewNoopProgressBar() ProgressLogger {
+	return &ProgressBarNoTTYLogger{}
+}
+
+// ProgressBarLogger display progress bar on output
 type ProgressBarLogger struct {
-	ctx                context.Context
 	cancelFunc         context.CancelFunc
 	bar                *pb.ProgressBar
 	ui                 goui.UI
 	finalMessage       string
 	errorMessagePrefix string
 }
 
-func (l *ProgressBarLogger) Start(progressChan <-chan regv1.Update) {
+// Start the display of the Progress Bar
+func (l *ProgressBarLogger) Start(ctx context.Context, progressChan <-chan regv1.Update) {
+	ctx, cancelFunc := context.WithCancel(ctx)
+	l.cancelFunc = cancelFunc
```

Comment:
im assuming we are not using ProgressBarLogger concurrently?

Reply:
No we are not using